web/src/popover_menus.js

```diff
@@ -256,6 +273,140 @@ export function initialize() {
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-mark-topic-read", (e) => {
+                const stream_id = topic_popover_stream_id(e);
+                if (!stream_id) {
+                    return;
+                }
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                unread_ops.mark_topic_as_read(stream_id, topic);
+                e.stopPropagation();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-delete-topic-messages", (e) => {
+                const stream_id = topic_popover_stream_id(e);
+                if (!stream_id) {
+                    return;
+                }
+
+                const topic = $(e.currentTarget).attr("data-topic-name");
+                const args = {
+                    topic_name: topic,
+                };
+
+                const html_body = render_delete_topic_modal(args);
+
+                confirm_dialog.launch({
+                    html_heading: $t_html({defaultMessage: "Delete topic"}),
+                    help_link: "/help/delete-a-topic",
+                    html_body,
+                    on_click() {
+                        message_edit.delete_topic(stream_id, topic);
+                    },
+                });
+
+                e.stopPropagation();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-toggle-resolved", (e) => {
+                const $topic_row = $(e.currentTarget);
+                const stream_id = Number.parseInt($topic_row.attr("data-stream-id"), 10);
+                const topic_name = $topic_row.attr("data-topic-name");
+                message_edit.with_first_message_id(stream_id, topic_name, (message_id) => {
+                    message_edit.toggle_resolve_topic(message_id, topic_name);
+                });
+
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            $popper.one("click", ".sidebar-popover-move-topic-messages", (e) => {
+                const $topic_row = $(e.currentTarget);
+                const stream_id = Number.parseInt($topic_row.attr("data-stream-id"), 10);
+                const topic_name = $topic_row.attr("data-topic-name");
+
+                stream_popover.build_move_topic_to_stream_popover(stream_id, topic_name);
+                e.stopPropagation();
+                e.preventDefault();
+                instance.hide();
+            });
+
+            new ClipboardJS($popper.find(".sidebar-popover-copy-link-to-topic")[0]).on(
+                "success",
+                () => {
+                    instance.hide();
+                },
+            );
+        },
+        onHidden(instance) {
+            instance.destroy();
+            topic_sidebar_popover_displayed = false;
```

Comment:
Just wanted to give you a heads up that we are aiming to clear the instance in the `popover_instances` object once it has been hidden, so please make sure to keep that in mind while working on this.

Reply:
Migrated this logic to use more common logic to clear the instance.